irr/src/CIrrDeviceSDL.cpp

```diff
@@ -1311,142 +1366,135 @@ void CIrrDeviceSDL::createKeyMap()
 	// the lookuptable, but I'll leave it like that until
 	// I find a better version.
 
-	KeyMap.reallocate(105);
-
 	// buttons missing
 
-	// Android back button = ESC
-	KeyMap.push_back(SKeyMap(SDLK_AC_BACK, KEY_ESCAPE));
-
-	KeyMap.push_back(SKeyMap(SDLK_BACKSPACE, KEY_BACK));
-	KeyMap.push_back(SKeyMap(SDLK_TAB, KEY_TAB));
-	KeyMap.push_back(SKeyMap(SDLK_CLEAR, KEY_CLEAR));
-	KeyMap.push_back(SKeyMap(SDLK_RETURN, KEY_RETURN));
+	KeyMap.emplace(SDLK_BACKSPACE, KEY_BACK);
```

Comment:
what I still don't understand is why are we mapping keycodes here? isn't this supposed to be scancodes?

Reply:
This is used for `IrrlichtDevice::getKeyFromScancode` which the key change menu uses to present keys to users. (Ideally the latter/`KeyPress` should use SDL keycodes directly instead of such a map, but we currently cannot do that as we discussed earlier.)

Reply:
I see. Please write this down in a comment.

Reply:
7ed10fb3e